File: briar-api/src/main/java/org/briarproject/briar/api/telegram/TelegramAuthSession.java
package org.briarproject.briar.api.telegram;

import org.briarproject.nullsafety.NotNullByDefault;

@NotNullByDefault
public interface TelegramAuthSession {

	enum RecoverableErrorDetail {
		NONE,
		MISSING_TDLIB,
		INVALID_IDENTIFIER
	}

	TelegramAuthState getCurrentState();

	RecoverableErrorDetail getRecoverableErrorDetail();

	void start();

	void submitIdentifier(String identifier);

	void submitCode(String code);

	void submitPassword(String password);

	void close();
}

File: briar-api/src/main/kotlin/org/briarproject/briar/api/telegram/TelegramAuthSession.kt
package org.briarproject.briar.api.telegram

interface TelegramAuthSession {
	enum class RecoverableErrorDetail {
		NONE,
		MISSING_TDLIB,
		INVALID_IDENTIFIER,
		INVALID_CODE,
		INVALID_PASSWORD
	}

	fun getCurrentState(): TelegramAuthState
	fun getRecoverableErrorDetail(): RecoverableErrorDetail
	fun start()
	fun submitIdentifier(identifier: String)
	fun submitCode(code: String)
	fun submitPassword(password: String)
	fun close()
}
